example/test/react_test_components.dart

```diff
@@ -282,9 +282,9 @@ class _NewContextProviderComponent extends react.Component2 {
   }
 
   render() {
-    Map provideMap = {'renderCount': this.state['renderCount']};
+    Map<String, dynamic> provideMap = {'renderCount': this.state['renderCount']};
```

Comment:
I'm not 100% sure for the reason this was changed, but the `<String, dynamic>` typing can be inferred:

```suggestion
    final provideMap = {'renderCount': this.state['renderCount']};
```

Reply:
This is changed because its inferred as <String, int> but the example changes the `value` for the "complex" values to <String, Map<String, dynamic>> and since its casted to int it will throw if you don't initially type it as <String, dynamic>

Reply:
@kealjones-wk I'm  not seeing this being inferred as `<String, int>`...

![image](https://user-images.githubusercontent.com/1750797/68209372-7aac5a80-ff90-11e9-8ce8-25ff5c16f389.png)

Reply:
at runtime in dart2js

Reply:
That actually seems like a Dart bug... We should report this if it hasn't been reported already.

Reply:
Sorry, so this issue was ACTUALLY due to providing a generic type to `createContext` that was typed as `Map<String, dynamic>` and then using Map as the type, it used `Map<dynamic, dynamic>` hence the issue i ran into, fix was to type these two vars to finals and then update the generic to just be `Map`